stl/inc/algorithm

```diff
@@ -4589,6 +4589,30 @@ namespace ranges {
             _STL_INTERNAL_STATIC_ASSERT(sentinel_for<_Se, _It>);
             _STL_INTERNAL_STATIC_ASSERT(indirect_binary_predicate<ranges::equal_to, projected<_It, _Pj>, const _Ty*>);
 
+#if _USE_STD_VECTOR_ALGORITHMS
+#if _HAS_CXX20
+            if constexpr (_Vector_alg_in_find_is_safe<_It, _Ty> && sized_sentinel_for<_Se, _It>
+                          && is_same_v<_Pj, identity>) {

```

Comment:
This appears to be hyper-bogus. The "eligible for optimization" check shouldn't be guarded. Additionally, this is ranges code, so C++20 is guaranteed.